packages/helix-shared-config/src/ValidationError.js

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2018 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+class ValidationError extends Error {
+  constructor(msg, errors = []) {
+    function prettyname(path, schema) {
+      if (path) {
+        if (path.startsWith('.strains')) {
+          return `${schema.title || 'Invalid Strain'} ${path.replace(/\.strains(\.|\[')(.*)/, '$2').replace(/'.*/, '')}`;
+        }
+        return `${schema.title || schema.$id} ${path}`;
+      }
+      return `${schema.title || schema.$id}`;
+    }
+
+    const detail = errors.map(({
+      keyword, dataPath, message, data, params, parentSchema,
+    }) => {
+      if (keyword === 'additionalProperties') {
+        return `${prettyname(dataPath, parentSchema)} has unknown property '${params.additionalProperty}'`;
+      }
+      if (keyword === 'required' && dataPath === '') {
+        return 'A set of strains and a default strain are missing.';
```

Comment:
hmm, this is rather specific for helix-config. but makes no sense for other schemas